### PR TITLE
#NP-27814 https://unit.atlassian.net/browse/NP-27814

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java [hunk 1]
```diff
@@ -93,20 +102,19 @@ private UUID toUuid(URI customerId) {
...
+        var objectMapper = new ObjectMapper(jsonFactory)
+                               .registerModule(new ProblemModule())
+                               .registerModule(new JavaTimeModule())
+                               .registerModule(new Jdk8Module())
+                               .enable(DeserializationFeature.ACCEPT_EMPTY_STRING_AS_NULL_OBJECT)
+                               .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+                               .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+                               .setSerializationInclusion(Include.ALWAYS);
+        return objectMapper.disable(SerializationFeature.INDENT_OUTPUT);
+        //return prettyJson
```
Comment (torarnet) [inline customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java:116]: ?

File: customer/src/test/java/no/unit/nva/customer/get/GetCustomerDoiHandlerTest.java [hunk 2]
```diff
@@ -70,6 +72,7 @@ void handleRequestReturnsOkWhenARequestWithAnExistingIdentifier() throws IOExcep
 
         assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
         assertThat(doiAgentResponse, is(equalTo(secretDto)));
+        assertTrue(response.getBody().contains("password\":null"));
```
Comment (torarnet) [inline customer/src/test/java/no/unit/nva/customer/get/GetCustomerDoiHandlerTest.java:75]: null?

**Review by torarnet — DISMISSED** [review]

File: customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java (hunk 1 above)
Comment (StigNorland) [reply customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java]: fixed

File: customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java [hunk 3]
```diff
@@ -93,20 +102,19 @@ private UUID toUuid(URI customerId) {
...
-    //                               .enable(DeserializationFeature.ACCEPT_EMPTY_STRING_AS_NULL_OBJECT)
-    //                               .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
-    //                               .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
-    //                               .setSerializationInclusion(Include.ALWAYS);
-    //        return objectMapper.disable(SerializationFeature.INDENT_OUTPUT);
-    //        //return prettyJson
-    //        //           ? objectMapper.enable(SerializationFeature.INDENT_OUTPUT)
-    //        //           : objectMapper.disable(SerializationFeature.INDENT_OUTPUT);
-    //    }
+    private ObjectMapper createJsonParser() {
```
Comment (brinxmat) [inline customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java:105]: Can you use the object mapper from commons?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java (hunk 3 above)
Comment (StigNorland) [reply customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java:105]: Frontend asked for all fields, even if empty, is this a breach on policies?

Comment (StigNorland) [reply customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java]: But yes, I didn't realize that I could configure mapper after creation.

Comment (StigNorland) [reply customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java:106]: I have tried and failed, error makes no sense....

File: customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java [hunk 4]
```diff
@@ -116,7 +116,34 @@ void handleInsertRequestOK()
...
                                    String.class);
-        var doiAgentResponse =  DoiAgentDto.fromJson(response.getBody());
+        var doiAgentResponse = DoiAgentDto.fromJson(response.getBody());
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_OK)));
+        assertThat(expectedDoiAgent, is(equalTo(doiAgentResponse)));
+    }
+
+    @Test
+    void handleUpdateUserNameRequestOK()
```
Comment (brinxmat) [inline customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java:126]: shouldAllowUserNameWhenUserNameUpdateIsRequestedByAuthorizedUser ?

File: customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java [hunk 5]
```diff
@@ -59,16 +59,18 @@ public void beforeEach() {
         existingCustomer = CustomerDataGenerator.createSampleCustomerDao().toCustomerDto();
     }
 
+    /**
+     * Task.
+     * <a href="https://unit.atlassian.net/browse/NP-27814">NP-27814</a>
+     */
```
Comment (brinxmat) [inline customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java:65]: should Allow Updates To What When

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java [hunk 6]
```diff
@@ -123,7 +123,7 @@ void handleInsertRequestOK()
     }
 
     @Test
-    void handleUpdateUserNameRequestOK()
+    void shouldPersistUserNameAsNull()
```
Comment (brinxmat) [inline customer/src/test/java/no/unit/nva/customer/update/UpdateCustomerDoiHandlerTest.java:126]: When?

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**